components/sst_importer/src/sst_importer.rs

```diff
@@ -657,40 +662,52 @@ impl ImportDir {
         Ok(path.save.exists())
     }
 
-    fn ingest<E: KvEngine>(
+    fn validate(
         &self,
         meta: &SstMeta,
-        engine: &E,
         key_manager: Option<Arc<DataKeyManager>>,
     ) -> Result<SSTMetaInfo> {
-        let start = Instant::now();
         let path = self.join(meta)?;
-        let cf = meta.get_cf_name();
-
-        // now validate the SST file.
         let path_str = path.save.to_str().unwrap();
-        let env = get_encrypted_env(key_manager.clone(), None /*base_env*/)?;
+        let env = get_encrypted_env(key_manager, None /*base_env*/)?;
         let env = get_inspected_env(Some(env))?;
-
         let sst_reader = RocksSstReader::open_with_env(&path_str, Some(env))?;
         sst_reader.verify_checksum()?;
+        // TODO: check the length and crc32 of ingested file.
         let meta_info = sst_reader.sst_meta_info(meta.to_owned());
+        Ok(meta_info)
+    }
 
-        super::prepare_sst_for_ingestion(&path.save, &path.clone, key_manager.as_deref())?;
-        let length = meta.get_length();
+    fn ingest<E: KvEngine>(
+        &self,
+        metas: &[SstMeta],
+        engine: &E,
+        key_manager: Option<Arc<DataKeyManager>>,
+    ) -> Result<()> {
+        let start = Instant::now();
 
-        // TODO check the length and crc32 of ingested file.
-        engine.reset_global_seq(cf, &path.clone)?;
-        IMPORTER_INGEST_BYTES.observe(length as _);
+        let mut paths = HashMap::new();
+        let mut ingest_bytes = 0;
+        for meta in metas {
+            let path = self.join(meta)?;
+            let cf = meta.get_cf_name();
+            super::prepare_sst_for_ingestion(&path.save, &path.clone, key_manager.as_deref())?;
+            ingest_bytes += meta.get_length();
+            engine.reset_global_seq(cf, &path.clone)?;
+            paths.entry(cf).or_insert_with(Vec::new).push(path);
+        }
 
         let mut opts = E::IngestExternalFileOptions::new();
         opts.move_files(true);
-        engine.ingest_external_file_cf(cf, &opts, &[path.clone.to_str().unwrap()])?;
-
+        for (cf, cf_paths) in paths {
+            let files: Vec<&str> = cf_paths.iter().map(|p| p.clone.to_str().unwrap()).collect();
+            engine.ingest_external_file_cf(cf, &opts, &files)?;
+        }
+        IMPORTER_INGEST_BYTES.observe(ingest_bytes as _);
```

Comment:
Better add some metric(histogram) to observe that how many files are ingested in once method.